kubernetes-extension/src/test/java/org/ballerinax/kubernetes/handlers/IstioGatewayGeneratorTests.java

```diff
@@ -0,0 +1,120 @@
+import org.ballerinax.kubernetes.models.istio.IstioGatewayModel;
+import org.ballerinax.kubernetes.models.istio.IstioPortModel;
+import org.ballerinax.kubernetes.models.istio.IstioServerModel;
+import org.junit.Assert;
+import org.junit.Test;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.ballerinax.kubernetes.KubernetesConstants.ISTIO_GATEWAY_FILE_POSTFIX;
+import static org.ballerinax.kubernetes.KubernetesConstants.YAML;
+
+/**
+ * Unit test cases for istio gateway models.
+ *
+ * @since 0.983.0
+ */
+public class IstioGatewayGeneratorTests {
+    @Test
+    public void testSimpleGateway() {
+        IstioGatewayModel istioGatewayModel = new IstioGatewayModel();
+        istioGatewayModel.setName("my-gateway");
+        istioGatewayModel.setNamespace("ballerina");
+        
+        Map<String, String> selectors = new LinkedHashMap<>();
+        selectors.put("app", "my-gatweway-controller");
+        istioGatewayModel.setSelector(selectors);
+        
+        List<IstioServerModel> serverModels = new LinkedList<>();
+        
+        // First server
+        IstioServerModel serverModel = new IstioServerModel();
+        IstioPortModel portModel = new IstioPortModel();
+        portModel.setNumber(80);
+        portModel.setProtocol("HTTP");
+        portModel.setName("http");
+        serverModel.setPort(portModel);
+        Set<String> hostModels = new LinkedHashSet<>();
+        hostModels.add("uk.bookinfo.com");
+        hostModels.add("eu.bookinfo.com");
+        serverModel.setHosts(hostModels);
+        IstioServerModel.TLSOptions tlsOptions = new IstioServerModel.TLSOptions();
+        tlsOptions.setHttpsRedirect(true);
+        serverModel.setTls(tlsOptions);
+        serverModels.add(serverModel);
+        
+        istioGatewayModel.setServers(serverModels);
+        
+        KubernetesContext.getInstance().getDataHolder().addIstioGatewayModel("sample-svc", istioGatewayModel);
+        try {
+            new IstioGatewayHandler().createArtifacts();
+            File yamlFile = new File("target" + File.separator + "kubernetes" + File.separator +
+                                     "hello" + ISTIO_GATEWAY_FILE_POSTFIX + YAML);
+            Yaml yaml = new Yaml();
+            Map gateway = (LinkedHashMap) yaml.load(FileUtils.readFileToString(yamlFile));
+    
+            // metadata
+            Map metadata = (Map) gateway.get("metadata");
+            Assert.assertEquals("my-gateway", metadata.get("name"));
+            Assert.assertEquals("ballerina", metadata.get("namespace"));
+    
+            Map spec = (Map) gateway.get("spec");
+            
+            Map selector = (Map) spec.get("selector");
+            Assert.assertEquals("my-gatweway-controller", selector.get("app"));
```

Comment:
Shall we use `KubernetesConstants.KUBERNETES_SELECTOR_KEY` here?

Reply:
Fixed with https://github.com/ballerinax/kubernetes/pull/220/commits/5f8887c1388816ebe18d4066c627a7d43a88bea9